docs/data_pipeline_procedure.md

```diff
@@ -1,5 +1,17 @@
 # Data Pipeline Procedure
 
+## Set working directory to default group
+   * Set working directory and its subdirectories to the default group
+      ```commandline
+      file_group=kbase
+      working_dir=dir_to_set_group
+      find /global/cfs/cdirs/kbase/collections/$working_dir -type d -exec chgrp $file_group {} \;
```

Comment:
why the braces and backslash?

Reply:
This looks like it sets the group on the directories but not on any existing files. The next command sets the sticky group bit but I don't think that affects any current files either...? Or am I wrong about that

Reply:
* it's taking the result of `find /global/cfs/cdirs/kbase/collections/$working_dir -type d` and execute each directories. 
* It only sets the directories and will only affect newly created files/sub-directories moving forward. If needed, we can manually sets the user's group for current files. But that will take quite some time since we do have a lot files. And we almostly only have CI files now.

Reply:
yeah, I get what the command does as a whole, I just don't understand the need for the `{} \` part

What I'm thinking about re current files is that the tools create directories when they run, so any files in those directories will be unreadable to anyone except the person that started the job (right?). Same for the downloaders.

Reply:
> chgrp and chmod needs to get the dir path returned from find.

ok,, I'm assuming that's what the braces are for but I still don't understand the need for the backslash - that's usually a line continuation marker but there's no line continuation here

re newly created files - ok now I get it. Wasn't thinking that directories would inherit the sticky bit

Reply:
the `\`  and `;` are together. Otherwise, `-exec` won't terminate properly. At least in NERSC it won't. 

(base) tgu@login25 collections  $ find /global/cfs/cdirs/kbase/collections/$working_dir -type d -exec chgrp $file_group {}
find: missing argument to `-exec'
(base) tgu@login25 collections  $ find /global/cfs/cdirs/kbase/collections/$working_dir -type d -exec chgrp $file_group {} ;
find: missing argument to `-exec'
(base) tgu@login25 collections  $ find /global/cfs/cdirs/kbase/collections/$working_dir -type d -exec chgrp $file_group {} \;
chgrp: changing group of '/global/cfs/cdirs/kbase/collections/homology_service': Operation not permitted
chgrp: changing group of '/global/cfs/cdirs/kbase/collections/homology_service/CI': Operation not permitted
chgrp: changing group of '/global/cfs/cdirs/kbase/collections/homology_service/CI/archive': Operation not permitted